### DISCUSSION TITLE
Allow to login with API key instead of user/password auth through github

### DISCUSSION THREAD

**Discussion by yarikoptic** [dandi/dandi-archive#1707] [Ideas] [answered]
An old version of this is https://github.com/dandi/dandiarchive-legacy/issues/792 . At that point I have worked around by implementing login via user/password in our webshots.  Since Sept 2023 we use 2FA for the bot account.  So far I found no way to "trick" it, e.g. to produce OTP key to provide using some tool (e.g., [oathtool](https://www.nongnu.org/oath-toolkit/man-oathtool.html)).  That is why I am coming back to check on feasibility to login to web instance by providing API KEY.  Do you think it would be possible?  Meanwhile I am disabling updates of the https://github.com/dandi/dandi-api-webshots

**Comment by waxlamp** [comment, marked as answer]
Unfortunately I don't think this is possible. The API key is specifically for authenticating to endpoint calls, rather than creating login credentials for use by the browser.

I have some ideas for how we could make this easier, but they involve changes to how we think about login for the app. Would you like to meet so we can discuss it?
